File: compiler-group-traslator/src/test/resources/IntEqual.java
public class IntEqual {
    public static void main(String[] args) {
        StaticJavaLib.assertTrue(2 == 3 == false);
        StaticJavaLib.assertTrue(3 == 2 == false);
        StaticJavaLib.assertTrue(2 == 2 == true);

        StaticJavaLib.assertTrue(equal(2, 3) == false);
        StaticJavaLib.assertTrue(equal(3, 2) == false);
        StaticJavaLib.assertTrue(equal(2, 2) == true);
    }
    
    static boolean equal(int i, int j) {
        return i == j;
    }
}    
    

File: compiler-group-traslator/src/test/resources/c/IntEqual.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
bool equal(int i, int j) {
	void* RU = i == j;
	return RU;

	//set all vars to zero and have the TEMP variable for the return expression
}
int main(char* args) {
	vm = newVM();
	assertTrue(2 == 3 == false);
	assertTrue(3 == 2 == false);
	assertTrue(2 == 2 == true);
	assertTrue(equal(2, 3) == false);
	assertTrue(equal(3, 2) == false);
	assertTrue(equal(2, 2) == true);
	return 0;
}